runtime/configs/types/proto/connection.proto

```diff
@@ -0,0 +1,10 @@
+syntax = "proto3";
```

Comment:
I think consolidating this with `runtime/configs/proto/p2p_config.proto` would be cleaner. Wdyt?

Reply:
I agree that it would make more sense for `P2PConfig` and `ConnectionTypes` to be in the same package (as it was before my changes). There's a slight hangup though in the `runtime/defaults` package. When replacing `defaults.DefaultP2PIsEmptyConnectionType` with `defaults.DefaultP2PConnectionType` we need to import the `ConnectionType` enum, which causes a cyclic import (as `configs/config.go` is currently consuming `defaults`). I'm not sure there's much that can be done in the way of consolidation so long as `configs` is importing `defaults` and `config.go` shares a package with the `ConnectionType` generated proto code. We could, of course, move more of the `runtime/configs` package into the new one (e.g. `P2PConfig`). The most aggressive version of this would result in separating the all the generated protobuf into the new package. While I think, in principal, this is probably the right direction to go, it would require many more changes to account for imports of other configs proto types.

Another alternative would be to consider moving (at least) the `ConnectionType` enum to the shared protobuf files (e.g. `shared/types/proto`) as this would necessarily change its package but makes less sense in my opinion as I don't think it's a fair to characterize this enum as "shared" based on current usage.

Reply:
I'm also very open to suggestions on better names for both the go and proto package (`types` and `conn`, respectively, as of 33aaafd).

Reply:
Appreciate the explanation @bryanchriswhite. I understand the point around consolidation and would say that lets just keep it as you have it.

The primary goal of this issue of using enums has been achieved relatively cleanly, and we'll have a lot more opportunities to clean up the code in the future.

The name lgtm!